### PR TITLE
MAT-8914 access locked library

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-8914](https://jira.cms.gov/browse/MAT-8914)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Review by etan-sb — APPROVED** [review]

File: src/components/editCqlLibrary/EditCqlLibrary.tsx [hunk 1]
```diff
@@ -239,10 +239,11 @@ const EditCqlLibrary = () => {
...
   const { updateRouteHandlerState } = routeHandlerStore;
-  const canEdit = checkUserCanEdit(
-    loadedCqlLibrary?.librarySet?.owner,
-    loadedCqlLibrary?.librarySet?.acls
-  );
+  const canEdit =
+    checkUserCanEdit(
+      loadedCqlLibrary?.librarySet?.owner,
+      loadedCqlLibrary?.librarySet?.acls
+    ) && !loadedCqlLibrary?.cqlLibraryLock;
```
Comment (etan-sb) [inline src/components/editCqlLibrary/EditCqlLibrary.tsx]: Do you need to add a feature flag check too?:

`&& !(featureFlags?.Locking && loadedCqlLibrary?.cqlLibraryLock);`

**Review by etan-sb — CHANGES_REQUESTED** [review]

File: src/components/editCqlLibrary/EditCqlLibrary.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/editCqlLibrary/EditCqlLibrary.tsx]: @etan-sb Thanks a lot! I just added it.

**Review by etan-sb — APPROVED** [review]
